lib/ddtrace/workers.rb

```diff
@@ -7,10 +7,13 @@ module Workers
     # will perform a task at regular intervals. The thread can be stopped
     # with the +stop()+ method and can start with the +start()+ method.
     class AsyncTransport
-      def initialize(interval, transport, buff_size, &task)
+      def initialize(interval, transport, queues, buff_size, &task)
         @task = task
         @interval = interval
-        @buffer = TraceBuffer.new(buff_size)
+        @buffers = {}
+        queues.each do |queue|
```

Comment:
we can make the code less generic because this ``AsyncTransport`` will only flush spans and services. In that case I will simply create two buffers like:
```
@traces_queue = ...
@services_queue = ...
```

Feel free to choose the right names.